spec/ui/editor_spec.moon

```diff
@@ -281,6 +281,25 @@ describe 'Editor', ->
       editor.buffer = buffer2
       assert.equal 15, cursor.pos
 
+    it 'updates .last_shown for buffer switched out', ->
+      time = os.time
+      now = time!
+      os.time = -> now
+      pcall ->
+        editor.buffer = Buffer {}
+      os.time = time
+      assert.same now, buffer.last_shown
+
+  context 'previewing', ->
+    it 'does not update last_shown for previewed or switched out buffer', ->
```

Comment:
Why would we not want to update `last_shown` for an ordinary buffer that is switched out as the result of a preview being shown?